### PR TITLE
Rewrite experiment_search_events_live_v1 materialized views without TO_JSON_STRING

### PR DESCRIPTION
**Pull request by scholtzan**
Using the workaround for `TO_JSON_STRING` of generating an array of indices works well for our `experiment_search_events_live_v1` materialized views. If I do the same for the `experiment_events_live_v1` materialized views I still run into the issue where `branch` is set to the `experiment` slug. So for now we have to keep using `TO_JSON_STRING` there.

For reference, this is basically the query I tried (shortened here) that returned the incorrect result:
```sql
CREATE MATERIALIZED VIEW
  AS
  SELECT
      submission_timestamp AS `timestamp`,
      event.category AS `type`,
      CAST(event.extra[safe_offset(i)].value as STRING) AS branch,  -- still set to the experiment slug
      CAST(event.extra[safe_offset(j)].value as STRING) AS experiment,
      event.name AS event_method
    FROM
      fenix_all_events,
      UNNEST(events) AS event,
      UNNEST(GENERATE_ARRAY(0, 100)) as i,
      UNNEST(GENERATE_ARRAY(0, 100)) as j
    WHERE
      event.category = 'nimbus_events' AND
      CAST(event.extra[safe_offset(i)].key as STRING) = 'branch' AND
      CAST(event.extra[safe_offset(j)].key as STRING) = 'experiment'
```

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql [hunk 1]
```diff
@@ -10,113 +10,52 @@ IF
...
-    -- (5)
-    SUM(CAST(JSON_EXTRACT_ARRAY(counts)[OFFSET(0)] AS INT64)) AS ad_clicks_count,
-    SUM(CAST(JSON_EXTRACT_ARRAY(counts)[OFFSET(1)] AS INT64)) AS search_with_ads_count,
-    SUM(CAST(JSON_EXTRACT_ARRAY(counts)[OFFSET(2)] AS INT64)) AS search_count,
+        -- Concatenating an element with value = 0 ensures that the count values are not null even if the array is empty
+        -- Materialized views don't support COALESCE or IFNULL
+    SUM(
+      CAST(
+        ARRAY_CONCAT(metrics.labeled_counter.browser_search_ad_clicks, [('', 0)])[
+          safe_offset(i)
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql:31]:
```suggestion
          SAFE_OFFSET(i)
```

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql [hunk 2]
```diff
@@ -10,113 +10,52 @@ IF
...
+        ].value AS int64
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql:32]:
```suggestion
        ].value AS INT64
```

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql [hunk 3]
```diff
@@ -10,113 +10,52 @@ IF
...
+      )
+    ) AS search_count
   FROM
-    aggregated_arrays,
-    UNNEST(JSON_EXTRACT_ARRAY(REPLACE(nested_counts, '"', ""))) AS counts
+    `moz-fx-data-shared-prod.org_mozilla_fenix_live.metrics_v1`
+  LEFT JOIN
+    UNNEST(ping_info.experiments) AS experiment,
+    -- Max. number of entries is around 10
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql:53]:
```suggestion
    UNNEST(ping_info.experiments) AS experiment
  CROSS JOIN
    -- Max. number of entries is around 10
```

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql [hunk 4]
```diff
@@ -10,113 +10,52 @@ IF
...
+    UNNEST(GENERATE_ARRAY(0, 50)) AS i
+  WHERE
+      -- Limit the amount of data the materialized view is going to backfill when created.
+      -- This date can be moved forward whenever new changes of the materialized views need to be deployed.
+    DATE(submission_timestamp) > '2021-04-20'
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_derived/experiment_search_events_live_v1/init.sql:58]: Update this date before merging.

**Review by jklukas — APPROVED** [review]
r+

I just have a few minor style comments.

**Comment by scholtzan** [comment]
Materialized views have been deployed

**Pull request merged**